heron/schedulers/src/java/com/twitter/heron/scheduler/mesos/MesosScheduler.java

```diff
@@ -0,0 +1,179 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.scheduler.mesos;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.apache.mesos.MesosSchedulerDriver;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+
+import com.twitter.heron.proto.scheduler.Scheduler;
+import com.twitter.heron.scheduler.mesos.framework.BaseContainer;
+import com.twitter.heron.scheduler.mesos.framework.MesosFramework;
+import com.twitter.heron.scheduler.mesos.framework.TaskUtils;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Constants;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Keys;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.scheduler.IScheduler;
+import com.twitter.heron.spi.utils.Runtime;
+import com.twitter.heron.spi.utils.SchedulerUtils;
+
+/**
+ * Schedule a topology to a mesos cluster
+ */
+
+public class MesosScheduler implements IScheduler {
+  private static final Logger LOG = Logger.getLogger(MesosScheduler.class.getName());
+
+  private Config config;
+  private Config runtime;
+  private MesosFramework mesosFramework;
+  private SchedulerDriver driver;
+
+  @Override
+  public void initialize(Config mConfig, Config mRuntime) {
+    this.config = mConfig;
+    this.runtime = mRuntime;
+    this.mesosFramework = getMesosFramework();
+  }
+
+  @Override
+  public void close() {
+    // Stop the SchedulerDriver
```

Comment:
No. Close() can be invoked when the jvm process SchedulerMain exits. There can be cases people restart SchedulerMain process to pick up earlier running topology (for fault tolerance). So the close of MesosScheduler should be independent with the killing of a topology.